data_safe_haven/commands/config.py

```diff
@@ -34,9 +35,29 @@ def upload(
 ) -> None:
     """Upload a configuration to the Data Safe Haven context"""
     context = ContextSettings.from_file().assert_context()
+    logger = LoggingSingleton()
+
+    # Create configuration object from file
     with open(file) as config_file:
         config_yaml = config_file.read()
     config = Config.from_yaml(config_yaml)
+
+    # Present diff to user
+    if Config.remote_exists(context):
+        if diff := config.remote_yaml_diff(context):
+            print("".join(diff))
```

Comment:
```suggestion
            for line in diff:
                logger.info(line)
```
I think it might be clearer to log each line separately